inaugurator/ceremony.py

```diff
@@ -313,20 +314,43 @@ def _additionalDownload(self, destination):
             downloadInstance.download(destination)
 
     def _makeSureDiskIsMountable(self):
-        if self._args.inauguratorTargetDeviceCandidate is None:
+        self._setTargetDevice()
+        self._createPartitionTable()
+        logging.info("Partitions created")
+        self._mountOp = mount.Mount()
+        assert self._bootPartitionPath is not None, "Please initialize boot partition path first"
+        self._mountOp.setBootPartitionPath(self._bootPartitionPath)
+
+    def _setTargetDevice(self):
+        if self._args.inauguratorTargetDeviceCandidate is not None:
+            logging.info("Explicit target devices indicated: %(deviceCandidates)s",
+                         dict(deviceCandidates=self._args.inauguratorTargetDeviceCandidate))
+            candidates = self._args.inauguratorTargetDeviceCandidate
+        elif self._args.inauguratorTargetDeviceType is not None:
             logging.info("Searching for target devices of type %(deviceType)s",
                          dict(deviceType=self._args.inauguratorTargetDeviceType))
             device = self._storageDevices.findFirstDeviceOfType(self._args.inauguratorTargetDeviceType,
                                                                 self._talkToServer)
+            if not device:
+                raise Exception("No %s devices found" % self._args.inauguratorTargetDeviceType)
             candidates = [device]
-        else:
-            candidates = self._args.inauguratorTargetDeviceCandidate
+        elif self._args.inauguratorTargetDeviceLabel is not None:
+            logging.info("Searching for target devices with label %(deviceLabel)s",
+                         dict(deviceLabel=self._args.inauguratorTargetDeviceLabel))
+            candidates = self._getDevicesWithLabel(self._args.inauguratorTargetDeviceLabel)
+            if not candidates:
+                raise Exception("No devices found with label %s" % self._args.inauguratorTargetDeviceLabel)
+
         self._targetDevice = targetdevice.TargetDevice.device(candidates)
-        self._createPartitionTable()
-        logging.info("Partitions created")
-        self._mountOp = mount.Mount()
-        assert self._bootPartitionPath is not None, "Please initialize boot partition path first"
-        self._mountOp.setBootPartitionPath(self._bootPartitionPath)
+
+    def _getDevicesWithLabel(self, label):
+        candidateList = []
+        output = sh.run("blkid")
+        for line in output.splitlines():
+            entry = line.strip().split(":", 1)
```

Comment:
checking on a live cluster:
/dev/mapper/inaugurator-temp: LABEL="temp" UUID="de711c76-99a2-4825-b9c6-956f8a9e58bb" TYPE="ext4" 
/dev/mapper/inaugurator-runtime_swap: LABEL="runtime_swap" UUID="6862f17f-ce77-41d6-b438-dbbcff932da7" TYPE="ext4" 
/dev/mapper/inaugurator-data: LABEL="data" UUID="11e5df1e-c6c9-4e11-82c9-820e61b03e6c" TYPE="ext4" 
/dev/mapper/docker-253:2-1048580-25257c2369d024e32045b64a2dc121933604d26c4daafa5a10112d12976a4a74: UUID="27fe0dc5-a035-4617-b247-dc94f6b926d7" TYPE="xfs" 
/dev/mapper/docker-253:2-1048580-26a46febce96d1ecf2c4e6637d1f4747beced6bf43e17fd4c754799bf82eb783: UUID="27fe0dc5-a035-4617-b247-dc94f6b926d7" TYPE="xfs" 


will ur split work ? Or you should not see docker mounts there?